test/avocado/selenium-base.py

```diff
@@ -7,27 +7,31 @@
 if not machine_test_dir in sys.path:
     sys.path.insert(1, machine_test_dir)
 
-from avocado import main
-from avocado.utils import process
-from seleniumlib import SeleniumTest, user, clickable, passwd
+from seleniumlib import SeleniumTest, user, clickable, passwd, visible
 
 class BasicTestSuite(SeleniumTest):
     """
     :avocado: enable
     """
     def test10Base(self):
-        out = process.run("hostname", shell=True)
+        self.wait_id('server-name')
+        self.error = False
+
+    def test15BaseSSHKeyAdded(self):
+        self.login()
+        self.logout()
```

Comment:
What does that do? There are no assertions after logging in. Does that have some side effect that the hostname comparison somehow relies on? (that would be weird). It seems that `server_element` is on the login page, so you don't actually need to log in, just open the cockpit URL?

Reply:
@martinpitt this is little bit tricky part how to "cleanly" transfer ssh public key to user account and do not rely that some infra does that.

without added ssh key I cannot use ``self.machine.execute`` to get ``hostname`` command, and ``self.login()`` function as last step does, that it add ssh pub key to user.
This is tricky part at least for this test, other tests are clean, because ``self.login`` is first function what was called.
Do you suggest better solution?

Reply:
That makes sense. I suggest to add a comment here to explain that.